log4j-api/src/main/java/org/apache/logging/log4j/util/PropertiesUtil.java

```diff
@@ -446,22 +447,36 @@ private static class Environment {
         private final Map<List<CharSequence>, String> tokenized = new ConcurrentHashMap<>();
 
         private Environment(final PropertySource propertySource) {
-            PropertyFilePropertySource sysProps = new PropertyFilePropertySource(LOG4J_SYSTEM_PROPERTIES_FILE_NAME, false);
-            try {
-                sysProps.forEach((key, value) -> {
-                    if (System.getProperty(key) == null) {
-                        System.setProperty(key, value);
-                    }
-                });
-            } catch (SecurityException ex) {
-                // Access to System Properties is restricted so just skip it.
-            }
+            PropertyFilePropertySource sysProps = new PropertyFilePropertySource(LOG4J_SYSTEM_PROPERTIES_FILE_NAME,
+                false);
+            sysProps.forEach((key, value) -> {
+                if (System.getProperty(key) == null) {
+                    System.setProperty(key, value);
+                }
+            });
             sources.add(propertySource);
+            sources.add(new SystemPropertiesPropertySource());
+            sources.add(new EnvironmentPropertySource());
```

Comment:
I wouldn't hard code these property sources. Isn't looking up services in the same protection domain always allowed?